modules/telemetry/telemetry/include/hephaestus/telemetry/struclog.h

```diff
@@ -0,0 +1,146 @@
+#pragma once
+
+#include <iomanip>
+#include <source_location>
+#include <sstream>
+#include <string>
+#include <string_view>
+#include <thread>
+#include <utility>
+#include <vector>
+
+namespace heph::telemetry {
+
+enum class Level : std::uint8_t { TRACE, DEBUG, INFO, WARN, ERROR, FATAL };
+auto operator<<(std::ostream& /*os*/, const Level& /*level*/) -> std::ostream&;
+
+template <typename T>
+concept NotQuotable = !std::convertible_to<T, std::string> && requires(std::ostream& os, T t) {
+  { os << t } -> std::same_as<std::ostream&>;
+};
+
+template <typename T>
+concept Quotable = std::convertible_to<T, std::string>;
+
+template <typename T>
+struct Field final {
+  std::string key;
+  T val;
+};
+
+/**
+ * @brief A class that allows easy composition of logs for structured logging.
+ *        Example(see also struclog.cpp):
+ *        namespace ht=heph::telemetry;
+ *        log(ht::LogEntry{Level::INFO, "my module", "adding"} | "id"_f(12345) | ht::Field{"tag", "test"}) ;
+ *
+ *        logs
+ *        'level=info hostname=goofy location=struclog.h:123 thread-id=5124 time=2023-12-3T8:52:02+0
+ * module="my module" message="adding" id=12345 tag="test"'
+ *
+ *        Remark: We would like to use libfmt with quoted formatting as proposed in
+ *                https://github.com/fmtlib/fmt/issues/825 once its included (instead of sstream).
+ *
+ */
+struct LogEntry {
+  using Fields = std::vector<Field<std::string>>;
+  using Clock = std::chrono::system_clock;
```

Comment:
You are touching a delicate point. We do not have documentation, simply because we are evolving fast and the cost of keeping docs updated at the moment outweigh the benefits. But we try to be consistent so I would say if you look at any other code that create type alias you would see we happen a `T`

Reply:
ok makes sense